File: Codeforces/global22/B.py

def inp():
    return list(map(int, input().split()))

for _ in range(int(input())):
    n, k = inp()
    a = inp()
    last = None
    ans = 'YES'
    for i in range(k - 2, -1, -1):
        e = a[i + 1] - a[i]
        if last is not None and e > last:
            ans = 'NO'
            break
        last = e

    if k > 1 and (n - k + 1) * last < a[0]:
        ans = 'NO'
    print(ans)

File: Codeforces/global22/B.cpp

#include <iostream>
#include <vector>
using namespace::std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n, k;
        cin >> n >> k;
        vector<int> a(k);
        for (int i=0; i < k; ++i)
            cin >> a.at(i);
        bool LAST = false;
        int last = 0;
        const char *ans = "YES";
        for (int i=k - 2; k >= 0; --k)
        {
            int e = a.at(i + 1) - a.at(i);
            if (LAST and e > last)
            {
                ans = "NO";
                break ;
            }
            last = e;
            LAST = true;
        }

        if (k > 1 and (n - k + 1) * last < a.at(0))
            ans = "NO";
        cout<<ans<<"\n";
    }
}
